server/src/test/java/org/elasticsearch/cluster/health/NoDataNodesHealthTests.java

```diff
@@ -0,0 +1,153 @@
+        state = allocationService.reroute(state, "Test_allocation");
+        state = allocationService.applyStartedShards(state, state.getRoutingNodes().shardsWithState(INITIALIZING));
+        IndexMetaData.Builder idxMetaBuilder = IndexMetaData.builder(state.metaData().index("TestIndex"));
+        for (final ShardRouting shards : state.getRoutingTable().index("TestIndex").shardsWithState(STARTED)) {
+            idxMetaBuilder.putInSyncAllocationIds(shards.getId(), Sets.newHashSet(shards.allocationId().getId()));
+        }
+        state = ClusterState.builder(state).metaData(MetaData.builder(state.metaData()).put(idxMetaBuilder)).build();
+        //asserting the cluster is in green after TestIndex Creation
+        assertEquals(ClusterHealthStatus.GREEN, new ClusterStateHealth(state).getStatus());
+        //Terminating data nodes
+        state = ClusterState.builder(state)
+                .nodes(DiscoveryNodes.builder(state.getNodes())
+                        .remove("node_d1").remove("node_d2").build())
+                .build();
+        //Removing dead nodes from cluster with a cluster reroute
+        state = allocationService.deassociateDeadNodes(state, true, "Test_allocation");
+        //asserting that a cluster state goes Red after data nodes goes terminated
+        assertEquals(ClusterHealthStatus.RED, new ClusterStateHealth(state).getStatus());
+        //Creating NewTestIndex meta deta
+        metaData = MetaData.builder(state.metaData())
+                .put(IndexMetaData.builder("NewTestIndex")
+                        .settings(settings(Version.CURRENT))
+                        .numberOfShards(2)
+                        .numberOfReplicas(0))
+                .build();
+        //changed cluster state
+        state = ClusterState.builder(state)
+                .metaData(metaData)
+                .routingTable(RoutingTable.builder(state.getRoutingTable()).addAsNew(metaData.index("NewTestIndex")).build()).build();
+        //allocation after newly created index
+        state = allocationService.reroute(state, "no data nodes");
+        assertEquals(ClusterHealthStatus.RED, new ClusterStateHealth(state).getStatus());
+        RoutingNodes routingNodes = state.getRoutingNodes();
+        RoutingNodes.UnassignedShards unassignedShards = routingNodes.unassigned();
+        assertFalse(unassignedShards.isEmpty());
+        RoutingNodes.UnassignedShards.UnassignedIterator unassignedIterator = unassignedShards.iterator();
+        while (unassignedIterator.hasNext()) {
+            ShardRouting shard = unassignedIterator.next();
+            UnassignedInfo shardInfo = shard.unassignedInfo();
+            /* asserting that the TestIndex shards have different AllocationStatus than DECIDERS_NO
+               and NewTestIndex status is DECIDERS_NO */
+            if (shard.getIndexName().equals("TestIndex")) {
+                assertNotEquals(shardInfo.getLastAllocationStatus(), UnassignedInfo.AllocationStatus.DECIDERS_NO);
```

Comment:
I think we don't really need to do these assertions and it'd be better to keep this test focussed on the cluster health.